torch/csrc/jit/runtime/symbolic_shape_registry.cpp

```diff
@@ -292,5 +292,14 @@ c10::optional<std::shared_ptr<Graph>> shapeComputeGraphForSchema(
   return c10::nullopt;
 }
 
+void RegisterShapeComputeGraphForSchema(const FunctionSchema& schema, std::shared_ptr<Graph> g) {
```

Comment:
nit: maybe take a `FunctionSchema*` since that's what's needed in the map? e.g. it might prevent someone from making an accidental copy if it's clear that the function takes a pointer as an argument.

also, just curious - do you know how the FunctionSchema lifetimes are handled? Seems like a lot of code for FunctionSchemas is written like this so clearly it works... is there a global cache somewhere that returns the same FunctionSchema object every time it sees the same schema?

Reply:
There is this comment about a static lifetime of operators, and corresponding static lifetime of function schema.. didnt look into it further yet but that file should b enough to figure things out

https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/operator.h#L36